crates/oxc_linter/src/context.rs

```diff
@@ -109,6 +109,25 @@ impl<'a> LintContext<'a> {
         self.semantic().scopes().ancestors(node.get().scope_id())
     }
 
+    #[must_use]
+    pub fn symbols(&self) -> &SymbolTable {
+        self.semantic().symbols()
+    }
+
+    #[must_use]
+    pub fn get_reference(&self, node: &AstNode<'a>) -> Option<&Reference> {
+        if let AstKind::IdentifierReference(reference) = node.get().kind() {
+            let id = self.nodes().get_node_id(node).unwrap();
+            let name = &reference.name;
+            let scope = self.scope(node);
+            scope.unresolved_references.get(name).and_then(|references| {
```

Comment:
is `unresolved_references` correctly implement now ?
I think `unresolved_references` is the references that doesn't have any defining occurrences of this variable but it was just pushed when a new `ReferenceIdentifier` appear?
https://github.com/Boshen/oxc/blob/main/crates/oxc_semantic/src/scope/builder.rs#L65

Reply:
It's more like "all the references" imo, among which unresolved ones don't have a `resolved_symbol_id`.